src/python/pants/option/global_options.py

```diff
@@ -1291,18 +1290,36 @@ def register_bootstrap_options(cls, register):
             "--use-deprecated-python-macros",
             advanced=True,
             type=bool,
-            default=True,
+            default=False,
             help=(
-                "If true, continue using Pants's deprecated macro system for "
+                "If true, use Pants's deprecated macro system for "
                 "`python_requirements`, `poetry_requirements`, and `pipenv_requirements` "
                 "rather than target generation.\n\n"
-                "The address for target generation is different. Rather than "
-                "`3rdparty/python:Django`, the address will look like `3rdparty/python#Django`. "
-                "The target generator (`python_requirements` et al) is a "
-                "target itself now, meaning you can give it a `name`. If the target generator "
+                "The address for macros is different. Rather than "
+                "`3rdparty/python#Django`, the address will look like `3rdparty/python:Django`. "
+                "The macro (`python_requirements` et al) also was not a proper target, meaning "
+                "that you could not give it a `name`. In contrast, if the target generator "
                 "sets its `name`, e.g. to `reqs`, generated targets will have an address like "
                 "`3rdparty/python:reqs#Django`."
             ),
+            removal_version="2.12.0.dev0",
```

Comment:
Maybe an aggressive timeline, but the upgrade is pretty simple thanks to `update-build-files`.

The motivation to removing the macros is:
1) We can update pantsd invalidation globs to not invalidate on pyproject.toml and requirements.txt
2) We can delete CAOFs I think :)